plugins/collections/list.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const listSchema = schema.models.collection.list;
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/collections',
+    config: {
+        description: 'Get collections for requesting user',
+        notes: 'Returns all collection records belonging to the requesting user',
+        tags: ['api', 'collections'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const { userFactory, collectionFactory } = request.server.app;
+            const { username } = request.auth.credentials;
+
+            return userFactory.get({ username })
+                .then((user) => {
+                    if (!user) {
```

Comment:
I'm not too sure but I was referencing how [LIST](https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/tokens/list.js#L30) was implemented for tokens and it had that check there so I wrote it as well. Not sure if it will directly error out if the user is not found.